File: common/src/main/kotlin/com/revolut/domain/repositories/ExchangeRepository.kt
package com.revolut.domain.repositories

import com.revolut.domain.models.Market
import com.revolut.domain.models.Ticker

/**
 * Created by yatsinar on 17/03/2018.
 * Revolut
 * All rights reserved
 */
expect class ExchangeRepository {

    suspend fun getAllMarkets(): List<Market>

    suspend fun getTicker(market: Market): Ticker

}

File: ios/mobius/data/ExchangeRepository.swift
//
//  ExchangeRepository.swift
//  Mobius Test App
//
//  Created by Иван Важнов on 24/03/2018.
//  Copyright © 2018 Revolut. All rights reserved.
//

import Foundation
import common

class ExchangeRepository: common.ExchangeRepository {
    
    private let api: BittrexApi
    
    init(api: BittrexApi) {
        self.api = api
    }
    
    override func getAllMarkets(callback: KotlinContinuation) {
        api.getAllMarkets(callback: callback)
    }
    
    override func getTicker(market: Market, callback: KotlinContinuation) {
        print("getTicker on thread is \(Thread.current)")
        api.getTicker(market: market, callback: callback)
    }
    
}
